tornado/test/websocket_test.py

```diff
@@ -329,6 +336,13 @@ def test_check_origin_invalid_subdomains(self):
 
         self.assertEqual(cm.exception.code, 403)
 
+    @gen_test
+    def test_handshake_response_headers(self):
+        ws = yield self.ws_connect('/handshake_response_headers')
+
+        self.assertEqual(ws.headers.get('Set-Cookie'), 'hello=1')
```

Comment:
Depends on what you're trying to do: `set_default_headers()` is called a second time on errors to reset the headers for the error page. I would probably use `prepare()` for things like this so the cookie only goes out on successful requests (and then we'd probably also need to call `self.clear()` before all the `set_status()` calls in `WebSocketHandler.get` so that any headers set during `prepare()` don't go out with the errors).